core/templates/dev/head/I18nFooter.ts

```diff
@@ -36,34 +36,36 @@ angular.module('oppia').directive('i18nFooter', [
             $http, $timeout, $translate, UserService,
             SUPPORTED_SITE_LANGUAGES) {
           var ctrl = this;
-          // Changes the language of the translations.
-          var preferencesDataUrl = '/preferenceshandler/data';
-          var siteLanguageUrl = '/save_site_language';
-          ctrl.supportedSiteLanguages = SUPPORTED_SITE_LANGUAGES;
+          this.$onInit = function() {
+            // Changes the language of the translations.
+            var preferencesDataUrl = '/preferenceshandler/data';
+            var siteLanguageUrl = '/save_site_language';
+            ctrl.supportedSiteLanguages = SUPPORTED_SITE_LANGUAGES;
 
-          // The $timeout seems to be necessary for the dropdown to show
-          // anything at the outset, if the default language is not English.
-          $timeout(function() {
-            // $translate.use() returns undefined until the language file is
-            // fully loaded, which causes a blank field in the dropdown, hence
-            // we use $translate.proposedLanguage() as suggested in
-            // http://stackoverflow.com/a/28903658
-            ctrl.currentLanguageCode = $translate.use() ||
-              $translate.proposedLanguage();
-          }, 50);
+            // The $timeout seems to be necessary for the dropdown
+            // to show anything at the outset, if the default language
+            // is not English.
+            $timeout(function() {
+              // $translate.use() returns undefined until the language
+              // file is fully loaded, which causes a blank field
+              // in the dropdown, hence we use $translate.proposedLanguage()
+              // as suggested in http://stackoverflow.com/a/28903658
+              ctrl.currentLanguageCode = $translate.use() ||
+                $translate.proposedLanguage();
+            }, 50);
 
-          ctrl.changeLanguage = function() {
-            $translate.use(ctrl.currentLanguageCode);
-            UserService.getUserInfoAsync().then(function(userInfo) {
-              if (userInfo.isLoggedIn()) {
-                $http.put(siteLanguageUrl, {
-                  site_language_code: ctrl.currentLanguageCode
-                });
-              }
-            });
+            ctrl.changeLanguage = function() {
```

Comment:
Here and elsewhere, why do you define everything in $onInit? I think functions should be defined outside $onInit, and only actions which need to be executed on initialization should be placed in onInit... based on https://docs.angularjs.org/guide/component#component-based-application-architecture

Reply:
I kept everything in onInt so that functions are also contained at the time of initialisation(if the function contains services which need to called on controllers) 
Do you feel its not a good practise to keep functions in onInt?

Reply:
I think calling functions in onInit is fine, but I'm not sure about the need to *define* functions in onInit. I'm not sure what you mean by your first sentence, could you please elaborate?

Reply:
Hi @seanlip 
as discussed i've improved the ussage of onInit PTAL this commit [here](https://github.com/oppia/oppia/pull/7502/commits/323cb95bc5a63f537922b6327f743bbd9885c2dd) Thanks!